src/main/java/io/hops/util/featurestore/ops/write_ops/FeaturestoreInsertIntoTrainingDataset.java

```diff
@@ -207,7 +203,7 @@ private void insertIntoHopsfsTrainingDataset(TrainingDatasetDTO trainingDatasetD
       }
       if(tfRecordSchemaJson != null){
         try {
-          FeaturestoreHelper.writeTfRecordSchemaJson(hopsfsTrainingDatasetDTO.getHdfsStorePath()
+          FeaturestoreHelper.writeTfRecordSchemaJson(trainingDatasetDTO.getLocation() + "/.."
```

Comment:
So the issue is that if you have a training dataset in `tfrecords` format you will end up with something like:
```
.../TrainingDatasets/training_ds_1
    ./tfschema
    ./training_ds
         ./actual data
```
The path that Hopsworks returns is `TrainingDataset/training_ds_1/training_ds`, so that you can feed it directly into Spark and get the data. 

As far as I could see we don't actually read the schema json when reading the training dataset in `tfrecords` format. 
Also, this nested dir for the training dataset only happens in the case of an HopsFS training dataset, and it doesn't happen if you are writing an external traininig dataset.


I'll check that the Python behaviour is correct.

Reply:
yes in that case double check the python code.

Reply:
Actually no, this code is wrong. I got confused. Hopsworks returns up to `training_ds_1` - then you should append the dataset name. I need to fix it.

Reply:
Fixed.